lib/twilioconnection.js

```diff
@@ -201,6 +252,9 @@ class TwilioConnection extends StateMachine {
         case 'bad':
           this._handleBad(message);
           break;
+        case 'busy':
```

Comment:
with the introduction of new states, we should add a check like this before handling messages. 

```
const legalMessages = {
  closed: [],
  connecting: ['bad', 'busy', 'welcome'], 
  early:  [], 
  open:  ['bad', 'bye', 'heartbeat', 'msg'],
  wait: [],
}

if (legalMessages[this.state].indexOf(message.type) < 0) {
  this._log.error(`Illegal message: ${message.type} when in state:${this.state}`);
  return;
}
```

Reply:
Thanks for making this comment, because I think it is worthwhile to consider which messages are allowed. Based upon what I have in C++, here is what I would use:

```
const legalMessages = {
  // All messages are allowed but ignored...
  closed: [],
  // msg is deferred in case it arrives out of order. Mostly a concern for transports that do not enforce ordering. Since WS does you could omit it
  connecting: ['bad', 'busy', 'welcome', 'msg'],
  // C++ defers welcome and msg but this seems unnecessary to me 
  early:  [], 
  open:  ['bad', 'bye', 'heartbeat', 'msg'],
  // I'd like to remove heartbeat but there was a server bug.
  // busy is possible due to the client sending up to 3x hellos and server responding to each hello
  wait: ['busy', 'heartbeat'],
}
```

Reply:
@manjeshbhargav WDYT?

Reply:
@ceaglest Sure, I'll add the log messages.

Reply:
@ceaglest @makarandp0 I'm adding these warning messages within their respective `_handle` methods, since I already check for invalid states there.